ge/org.osate.ge.ba/src/org/osate/ge/ba/ui/properties/EmbeddedTextEditor.java

```diff
@@ -24,41 +24,48 @@
 package org.osate.ge.ba.ui.properties;
 
 import org.eclipse.core.resources.IProject;
-import org.eclipse.jface.layout.GridDataFactory;
 import org.eclipse.jface.layout.GridLayoutFactory;
 import org.eclipse.swt.SWT;
 import org.eclipse.swt.custom.StyledText;
 import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.layout.GridData;
 import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Caret;
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.Control;
 import org.osate.ge.swt.SwtUtil;
 
-public class EmbeddedTextControls extends Composite {
-	private final StyledText styledText;
-	private final Button btn;
+public class EmbeddedTextEditor extends Composite {
+	private StyledText styledText;
+	private Button btn;
 	private EmbeddedXtextAdapter xtextAdapter;
+	private final int styledTextStyle;
+	private final GridData styledTextLayoutData;
 
-	public EmbeddedTextControls(final Composite parent, final int style, final int styledTextStyle, final int btnStyle,
-			final String editBtnText) {
+	public EmbeddedTextEditor(final Composite parent, final int style, final int styledTextStyle,
+			final GridData styledTextLayoutData) {
```

Comment:
Remove this styledTextLayoutData parameter. Internal detail. Using the same value in both cases. Just grab and fill contents of container.